src/rpc/misc.cpp

```diff
@@ -555,6 +556,61 @@ UniValue getmemoryinfo(const JSONRPCRequest& request)
     }
 }
 
+uint32_t getCategoryMask(UniValue cats) {
+    cats = cats.get_array();
+    uint32_t mask = 0;
+    for (unsigned int i = 0; i < cats.size(); ++i) {
+        uint32_t flag = 0;
+        std::string cat = cats[i].get_str();
+        if (!GetLogCategory(&flag, &cat)) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "unknown logging category " + cat);
+        }
+        mask |= flag;
+    }
+    return mask;
+}
+
+UniValue logging(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() > 2) {
+        throw std::runtime_error(
+            "logging [include,...] <exclude>\n"
+            "Gets and sets the logging configuration.\n"
+            "When called without an argument, returns the list of categories that are currently being debug logged.\n"
+            "When called with arguments, adds or removes categories from debug logging.\n"
+            "The valid logging categories are: " + ListLogCategories() + "\n"
+            "libevent logging is configured on startup and cannot be modified by this RPC during runtime."
+            "Arguments:\n"
+            "1. \"include\" (array of strings) add debug logging for these categories.\n"
+            "2. \"exclude\" (array of strings) remove debug logging for these categories.\n"
+            "\nResult: <categories>  (string): a list of the logging categories that are active.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("logging", "\"[\\\"all\\\"]\" \"[\\\"http\\\"]\"")
+            + HelpExampleRpc("logging", "[\"all\"], \"[libevent]\"")
+        );
+    }
+
+    if (request.params.size() > 0 && request.params[0].isArray()) {
+        logCategories |= getCategoryMask(request.params[0]);
+    }
+
+    if (request.params.size() > 1 && request.params[1].isArray()) {
+        logCategories &= ~getCategoryMask(request.params[1]);
+    }
+
+    // Update libevent logging. If the library version doesn't allow it, UpdateHTTPServerLogging() is a no-op.
+    UpdateHTTPServerLogging(logCategories & BCLog::LIBEVENT);
+
+    UniValue result(UniValue::VOBJ);
+    std::vector<CLogCategoryActive> vLogCatActive = ListActiveLogCategories();
+    for (const auto& logCatActive : vLogCatActive) {
+        result.pushKV(logCatActive.category, logCatActive.active);
+    }
+    result.pushKV("WARNING", "libevent logging cannot be updated when using libevent before v2.1.1.");
```

Comment:
I'm perfectly fine with removing this warning from this pull and doing that later. I just don't think it should be added unconditionally.

> I've already applied the include/excude bitmasks by this point, and we need the before/after state to know whether to throw.

Sure. `if((before ^ after) & BCLog::LIBEVENT)` would work :-) Though it's probably best to do the check inside UpdateHTTPServerLogging  because there you have the previous state, the desired new state, and the libevent information available.

> we only want to raise an error if only libevent logging is changed 

That's a good point, and indeed speaks for making it a warning instead.